### PR TITLE
add code in user.sls to create users from pillar

### PR DESCRIPTION
**Pull request by apbeers**
I added some code in the user.sls file that will create the user if it doesn't exist from the salt pillar section "samba_users". Let me know what you think!

### CODE REVIEW

File: pillar.example [hunk 1]
```diff
@@ -27,9 +27,19 @@ samba_sections:
     force group: sharegroup
     create mask: 0660
     directory mask: 2770
+  user1share:
```
Comment (gravyboat) [inline pillar.example]: This is a good addition. I would like to see this modified so it's all part of the `samba_users` section for adding users, that would also allow for a more simplified loop.

File: pillar.example [hunk 2]
```diff
@@ -27,9 +27,19 @@ samba_sections:
...
+    create mode: 0660
+    directory mode: 0770
+    public: no
+    writable: yes
+    printable: no
+
 
 samba_users:
-   user1: 
-     password: user1sambapassword
```
Comment (gravyboat) [inline pillar.example]: Could user details instead be here?

File: samba/users.sls [hunk 3]
```diff
@@ -4,6 +4,13 @@ include:
 {% endif %}
 
 {% for login,user in pillar.get('samba_users', {}).items() %}
+{{ login }}:
+  user.present:
+    - fullname: {{ login }}
+    - password: {{ user.password }}
+{% endfor %}
+
+{% for login,user in pillar.get('samba_users', {}).items() %}
```
Comment (gravyboat) [inline samba/users.sls]: Seems like this one is redundant if we fix the way users are structured.

**Comment by apbeers** [comment]
So are you wanting to create a share file for each user in samba_users? If so I could put a value like "create_share = True:False" in the samba_users.user section of pillar.

I also combined the FOR loops in users.sls

**Comment by gravyboat** [comment]
@apbeers I'm not sure, that's why I posed it as a question. My Samba knowledge is limited so I don't know if that makes sense to do or not. Do you think it makes sense? Is it easier/better for users of Samba?

**Comment by apbeers** [comment]
I don't really think so. The way I'm using this at work is to push updates to the config mostly. If I need to add a user to the share, it would be easier to just leave the users section how it is and add their username to the proper samba share using the pillar. 

Something else I think would be good is to automatically create the directory for a share from the config in the pillar if they specify a path. The only issue i've been having with this is that the "valid users" parameter has a space in it so I can't reference it from in any kind of jinja

**Comment by apbeers** [comment]
Even if we don't add anything else, creating the users from the pillar prevents salt from giving an error when you run it. This does make it nicer to use

**Comment by gravyboat** [comment]
Alright I'm fine with it then, if anyone has concerns they can make another PR or propose backing this out.

**Pull request merged**